### DISCUSSION TITLE
Extend _collections table

### DISCUSSION THREAD

**Discussion by skowrons** [pocketbase/pocketbase#5840] [Q&A] [answered]
Hi everyone,

does anybody know if it is safe to extend the _collections table in PocketBase by adding new columns, or is it recommended to create a separate wrapper table instead? What are the potential risks or best practices associated with directly modifying the _collections table?

**Comment by ganigeorgiev** [comment]
It is not safe because future system migrations may brake it.

What exactly is your use case?

**Reply by skowrons** [reply]
My goal is to enrich the collections with additional metadata like odata (https://docs.oasis-open.org/odata/odata-csdl-xml/v4.01/odata-csdl-xml-v4.01.html#sec_VocabularyandAnnotation). This would be a mighty extension point for data driven uis.

1. To improve collaboration within a larger team, I plan to add a description field to each entity. 
2. Furthermore, I want to leverage metadata, similar to the approach used in some OData frameworks, to provide hints to the frontend on how certain datasets should be rendered. For instance, UI5 by SAP includes a feature called SmartLink, which utilizes OData annotations to generate a preview of an object. Essentially, it defines which datasets should be displayed, offering a more dynamic and intuitive way to interact with data

Here the smart link popover:
![Bildschirmfoto 2024-11-25 um 17 32 41](https://github.com/user-attachments/assets/94878beb-8866-48bb-a103-2c480aaeb473)

**Reply by ganigeorgiev** [reply, marked as answer]
No, in addition to the risk that it may break in the future, the `_collections` table and its Web APIs are usually intended to be accessed only by PocketBase admins/superusers and as far as I understand based on the above screenshots part of the collection model configuration will have to be externally accessible.

You'll be better maintaining the required metadata in a separate structure outside of the `_collections` table.
